sympy/core/basic.py

```diff
@@ -1557,12 +1557,24 @@ def doit(self, **hints):
             return self
 
     def _eval_rewrite(self, pattern, rule, **hints):
+        from sympy.simplify.simplify import simplify
         if self.is_Atom:
             if hasattr(self, rule):
                 return getattr(self, rule)()
             return self
 
-        if hints.get('deep', True):
+        if rule == '_eval_rewrite_as_sinc':
```

Comment:
expr, having `sinc` can't be simplified. In the test case I was getting this : 

```
Out[9]: 
⎧                                            1.0                                               for π⋅k = 0
⎪                                                                                                         
⎨ⅈ⋅(-1.0⋅ⅈ⋅k⋅sinc(π⋅k) + (0.5⋅k - 0.25)⋅sinc(π⋅(k - 1/2)) + (0.5⋅k + 0.25)⋅sinc(π⋅(k + 1/2)))             
⎪────────────────────────────────────────────────────────────────────────────────────────────   otherwise 
⎩                                             k                                                           

```

So I think it is better to first convert in to `sin` then `sinc`. 

I should use `trigsimp`  instead of `simplify`. I will edit it.